src/libsyntax/parse/token.rs

```diff
@@ -265,10 +265,24 @@ pub enum TokenKind {
 #[cfg(target_arch = "x86_64")]
 static_assert_size!(TokenKind, 16);
 
+#[derive(Clone, Copy, PartialEq, RustcEncodable, RustcDecodable, Debug)]
+pub enum IsJoint {
```

Comment:
This can almost reuse `proc_macro::Spacing`, except that it doesn't implement `Rustc{Encodable,Decodable}`.
Perhaps it can still be done if the `*codable` traits are implemented for `Token` manually?

Reply:
I definitely think that we should  at least rename this to Spacing. As for reusing the actual type, I don't know. In general, I feel that that a boilerplaty conversion on the boundary between two systems is better than sharing actual types, as it is more resilient to changes.

Reply:
At least `proc_macro::Spacing` is stable and won't change though.

Reply:
>except that it doesn't implement Rustc{Encodable,Decodable}

Actually the impls can be added to libserialize, like it's done for some libstd types used in the compiler in https://github.com/rust-lang/rust/blob/master/src/libserialize/collection_impls.rs.

Reply:
Anyway, at least the naming should be consistent with `proc_macro::Spacing` (and let's also remove `pub use IsJoint::*;`).